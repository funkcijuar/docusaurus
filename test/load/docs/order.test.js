import createOrder from '@lib/load/docs/order';

describe('createOrder', () => {
  test('should populate docs index from multiple sidebars', () => {
    const result = createOrder({
      docs: {
        Category1: ['doc1', 'doc2'],
        Category2: ['doc3', 'doc4']
      },
      otherDocs: {
        Category1: ['doc5']
      }
    });
    expect(result).toMatchSnapshot();
  });

  test('should resolve docs from older versions', () => {
    const result = createOrder({
      docs: {
        Category1: ['doc1']
      },
      'version-1.2.3-docs': {
        Category1: ['version-1.2.3-doc2'],
        Category2: ['version-1.2.3-doc1']
      }
    });
    expect(result).toMatchSnapshot();
  });

  test('edge cases', () => {
    expect(createOrder({})).toEqual({});
    expect(createOrder(null)).toEqual({});
    expect(createOrder(undefined)).toEqual({});
  });
});
