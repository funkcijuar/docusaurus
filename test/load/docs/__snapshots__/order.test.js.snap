// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createOrder should populate docs index from multiple sidebars 1`] = `
Object {
  "doc1": Object {
    "category": "Category1",
    "next": "doc2",
    "previous": undefined,
    "sidebar": "docs",
  },
  "doc2": Object {
    "category": "Category1",
    "next": "doc3",
    "previous": "doc1",
    "sidebar": "docs",
  },
  "doc3": Object {
    "category": "Category2",
    "next": "doc4",
    "previous": "doc2",
    "sidebar": "docs",
  },
  "doc4": Object {
    "category": "Category2",
    "next": undefined,
    "previous": "doc3",
    "sidebar": "docs",
  },
  "doc5": Object {
    "category": "Category1",
    "next": undefined,
    "previous": undefined,
    "sidebar": "otherDocs",
  },
}
`;

exports[`createOrder should resolve docs from older versions 1`] = `
Object {
  "doc1": Object {
    "category": "Category1",
    "next": undefined,
    "previous": undefined,
    "sidebar": "docs",
  },
  "version-1.2.3-doc1": Object {
    "category": "Category2",
    "next": undefined,
    "previous": "version-1.2.3-doc2",
    "sidebar": "version-1.2.3-docs",
  },
  "version-1.2.3-doc2": Object {
    "category": "Category1",
    "next": "version-1.2.3-doc1",
    "previous": undefined,
    "sidebar": "version-1.2.3-docs",
  },
}
`;
